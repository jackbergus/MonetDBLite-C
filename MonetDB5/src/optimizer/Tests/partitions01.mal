#testing the BAT partition manager
# use a fake sql library
module sql;
pattern bind(s:str,t:str,i:str,flag:int):bat[:oid,:int]
address MDBdummy;
pattern rsColumn(b:any...):void
address MDBdummy;
pattern resultSet(b:any...):void
address MDBdummy;
pattern exportResult(b:any...):void
address MDBdummy;

function query(A0:sht):void;
    x1:bat[:oid,:int]{bid=812,rows=1,notnil=true}  := sql.bind("sys","x","i",0);
    x6:bat[:oid,:int]{bid=808,rows=0,notnil=true}  := sql.bind("sys","x","i",1);
    constraints.emptySet(x6);
    x9:bat[:oid,:int]{bid=814,rows=0,notnil=true}  := sql.bind("sys","x","i",2);
    constraints.emptySet(x9);
    x21:bat[:oid,:int]{bid=815,rows=1,notnil=true}  := sql.bind("sys","x","j",1);
    x18{rows=1} := algebra.markT(x1,0@0);
    x19{rows=1} := bat.reverse(x18);
    x20{rows=1} := algebra.join(x19,x1);
    x28{rows=1} := algebra.join(x19,x21);
    x29 := sql.resultSet(2,1,x20);
    sql.rsColumn(x29,"sys.x","i","int",32,0,x20);
    sql.rsColumn(x29,"sys.x","j","int",32,0,x28);
    sql.exportResult(x29,"");
end query;
optimizer.partitions("sql","query");
mdb.list("sql","query");
