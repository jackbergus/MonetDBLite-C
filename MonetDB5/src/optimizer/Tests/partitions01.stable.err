stderr of test 'partitions01` in directory 'src/optimizer` itself:


# 09:48:30 >  
# 09:48:30 >  Mtimeout -timeout 60 mserver5 "--config=/ufs/mk/monet5/Linux/etc/monetdb5.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5//Linux/lib/MonetDB5:/ufs/mk/monet5//Linux/lib/MonetDB5/lib:/ufs/mk/monet5//Linux/lib/MonetDB5/bin" --set "gdk_dbfarm=/ufs/mk/monet5//Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/monet5//Linux/var/MonetDB5/sql_logs" --set "xquery_logdir=/ufs/mk/monet5//Linux/var/MonetDB5/xquery_logs" --set mapi_port=36708 --set xrpc_port=48585 --set monet_prompt= --trace --dbname=mTests_src_optimizer  partitions01.mal
# 09:48:30 >  

#warning: please don't forget to set your vault key!
#(see /ufs/mk/monet5/Linux/etc/monetdb5.conf)

# 09:48:30 >  
# 09:48:30 >  Done.
# 09:48:30 >  

