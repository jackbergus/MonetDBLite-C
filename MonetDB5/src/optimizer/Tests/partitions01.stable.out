stdout of test 'partitions01` in directory 'src/optimizer` itself:


# 09:48:30 >  
# 09:48:30 >  Mtimeout -timeout 60 mserver5 "--config=/ufs/mk/monet5/Linux/etc/monetdb5.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5//Linux/lib/MonetDB5:/ufs/mk/monet5//Linux/lib/MonetDB5/lib:/ufs/mk/monet5//Linux/lib/MonetDB5/bin" --set "gdk_dbfarm=/ufs/mk/monet5//Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/monet5//Linux/var/MonetDB5/sql_logs" --set "xquery_logdir=/ufs/mk/monet5//Linux/var/MonetDB5/xquery_logs" --set mapi_port=36708 --set xrpc_port=48585 --set monet_prompt= --trace --dbname=mTests_src_optimizer  partitions01.mal
# 09:48:30 >  

# MonetDB Server v5.0.0_beta2_1
# Copyright (c) 1993-2007 CWI, all rights reserved
# Compiled for x86_64-redhat-linux-gnu/64bit with 64bit OIDs dynamically linked
# dbname:mTests_src_optimizer
# Visit http://monetdb.cwi.nl/ for further information
##testing the BAT partition manager
## use a fake sql library
#module sql;
#pattern bind(s:str,t:str,i:str,flag:int):bat[:oid,:int]
pattern sql.bind(s:str,t:str,i:str,flag:int):bat[:oid,:int] address MDBdummy;	# 0  (bind:bat[:oid,:int])<-(s:str)(t:str)(i:str)(flag:int)
#pattern rsColumn(b:any...):void
pattern sql.rsColumn(b:any...):void address MDBdummy;	# 0  (rsColumn:void)<-(b:any)
#pattern resultSet(b:any...):void
pattern sql.resultSet(b:any...):void address MDBdummy;	# 0  (resultSet:void)<-(b:any)
#pattern exportResult(b:any...):void
pattern sql.exportResult(b:any...):void address MDBdummy;	# 0  (exportResult:void)<-(b:any)
#
#function user.query(A0:sht):void;
#   x2:bat[:oid,:int]{bid=777,rows=1,notnil=true}  := sql.bind("sys","tmp","i",0);
#    x11:bat[:oid,:int]{bid=313,rows=0,notnil=true}  := sql.bind("sys","tmp","i",1);
#    constraints.emptySet(x11);
#    x13:bat[:oid,:int]{bid=778,rows=0,notnil=true}  := sql.bind("sys","tmp","i",2);
#    constraints.emptySet(x13);
#    x19:bat[:oid,:int]{bid=779,rows=1,notnil=true}  := sql.bind("sys","tmp","j",0);
#    x7 := calc.int(A0);
#    x8{rows=1} := algebra.uselect(x2,nil:int,x7,false,false);
#    x16{rows=1} := algebra.markT(x8,0@0);
#    x17{rows=1} := bat.reverse(x16);
#    x18{rows=1} := algebra.join(x17,x2);
#    x21{rows=1} := algebra.join(x17,x19);
#    x22 := sql.resultSet(2,1,x18);
#    sql.rsColumn(x22,"sys.tmp","i","int",32,0,x18);
#    sql.rsColumn(x22,"sys.tmp","j","int",32,0,x21);
#    sql.exportResult(x22,"");
#end query;
function sql.query(A0:sht):void;	# 0  (query:void)<-(A0:sht)
    x1:bat[:oid,:int]{notnil=true,rows=1,bid=812}  := sql.bind("sys","x","i",0);	# 1 MDBdummy (x1:bat[:oid,:int])<-(_3:str)(_4:str)(_5:str)(_6:int)
    x6:bat[:oid,:int]{notnil=true,rows=0,bid=808}  := sql.bind("sys","x","i",1);	# 2 MDBdummy (x6:bat[:oid,:int])<-(_3:str)(_4:str)(_5:str)(_8:int)
    constraints.emptySet(x6);	# 3 CemptySet (_9:void)<-(x6:bat[:oid,:int])
    x9:bat[:oid,:int]{notnil=true,rows=0,bid=814}  := sql.bind("sys","x","i",2);	# 4 MDBdummy (x9:bat[:oid,:int])<-(_3:str)(_4:str)(_5:str)(_11:int)
    constraints.emptySet(x9);	# 5 CemptySet (_12:void)<-(x9:bat[:oid,:int])
    x21:bat[:oid,:int]{notnil=true,rows=1,bid=815}  := sql.bind("sys","x","j",0);	# 6 MDBdummy (x21:bat[:oid,:int])<-(_3:str)(_4:str)(_14:str)(_6:int)
    x18{rows=1} := algebra.markT(x1,0@0);	# 7 ALGtmark (x18:bat[:oid,:oid])<-(x1:bat[:oid,:int])(_16:oid)
    x19{rows=1} := bat.reverse(x18);	# 8 BKCreverse (x19:bat[:oid,:oid])<-(x18:bat[:oid,:oid])
    x20{rows=1} := algebra.join(x19,x1);	# 9 ALGjoin (x20:bat[:oid,:int])<-(x19:bat[:oid,:oid])(x1:bat[:oid,:int])
    x28{rows=1} := algebra.join(x19,x21);	# 10 ALGjoin (x28:bat[:oid,:int])<-(x19:bat[:oid,:oid])(x21:bat[:oid,:int])
    x29 := sql.resultSet(2,1,x20);	# 11 MDBdummy (x29:void)<-(_11:int)(_8:int)(x20:bat[:oid,:int])
    sql.rsColumn(x29,"sys.x","i","int",32,0,x20);	# 12 MDBdummy (_21:void)<-(x29:void)(_22:str)(_5:str)(_23:str)(_24:int)(_6:int)(x20:bat[:oid,:int])
    sql.rsColumn(x29,"sys.x","j","int",32,0,x28);	# 13 MDBdummy (_25:void)<-(x29:void)(_22:str)(_14:str)(_23:str)(_24:int)(_6:int)(x28:bat[:oid,:int])
    sql.exportResult(x29,"");	# 14 MDBdummy (_26:void)<-(x29:void)(_27:str)
end query;	# 15  
#optimizer.partitions("user","query");
#
function user.main():void;	# 0  (main:void)
#testing the BAT partition manager 	# 1  (_1:str)
# use a fake sql library 	# 2  (_2:str)
    mdb.list("sql","query");	# 3 MDBlist3 (_6:void)<-(_4:str)(_5:str)
end main;	# 4  
function sql.query{segment=2}(A0:sht):void;
    x1:bat[:oid,:int]{segment=2,notnil=true,rows=1,bid=812}  := sql.bind("sys","x","i",0);
    bpm2:bat[:oid,:int]  := bpm.take("sys_x_i");
    x6:bat[:oid,:int]{segment=-1,notnil=true,rows=0,bid=808}  := sql.bind("sys","x","i",1);
    constraints.emptySet(x6);
    x9:bat[:oid,:int]{segment=-1,notnil=true,rows=0,bid=814}  := sql.bind("sys","x","i",2);
    constraints.emptySet(x9);
    x21:bat[:oid,:int]{segment=5,notnil=true,rows=1,bid=815}  := sql.bind("sys","x","j",0);
    bpm13:bat[:oid,:int]  := bpm.take("sys_x_j");
    _34 := bpm.new(:oid,:oid);
barrier x1:bat[:oid,:int]{segment=2,notnil=true,rows=1,bid=812}  := bpm.newIterator(bpm2);
    x18{segment=2,rows=1} := algebra.markT(x1,0@0);
    x19{segment=2,rows=1} := bat.reverse(x18);
    bpm.addPartition(_34,x19);
    redo x1:bat[:oid,:int]{segment=2,notnil=true,rows=1,bid=812}  := bpm.hasMoreElements(bpm2);
exit x1:bat[:oid,:int]{segment=2,notnil=true,rows=1,bid=812} ;
    _40 := bpm.new(:oid,:int);
barrier x19{segment=2,rows=1} := bpm.newIterator(_34);
barrier x1:bat[:oid,:int]{segment=2,notnil=true,rows=1,bid=812}  := bpm.newIterator(bpm2);
    x20{segment=6,rows=1} := algebra.join(x19,x1);
    x28{segment=6,rows=1} := algebra.join(x19,x21);
    bpm.addPartition(_40,x28);
    redo x1:bat[:oid,:int]{segment=2,notnil=true,rows=1,bid=812}  := bpm.hasMoreElements(bpm2);
    redo x19{segment=2,rows=1} := bpm.hasMoreElements(_34);
exit x1:bat[:oid,:int]{segment=2,notnil=true,rows=1,bid=812} ;
exit x19{segment=2,rows=1};
    x20{segment=6,rows=1} := bpm.glue(_40);
    x29{segment=7} := sql.resultSet(2,1,x20);
    sql.rsColumn(x29,"sys.x","i","int",32,0,x20);
    x28{segment=6,rows=1} := bpm.glue(_40);
    sql.rsColumn(x29,"sys.x","j","int",32,0,x28);
    sql.exportResult(x29,"");
end query;

# 09:48:30 >  
# 09:48:30 >  Done.
# 09:48:30 >  

